classes/PipelineRequest/README.md

```diff
@@ -65,3 +65,59 @@ The error list defaults to an empty array.
 A callback that gets triggered when a request is sent.
 It expects references to the resolve and reject Promise callback functions.
 This method can be overridden by a child class.
+
+## Testing - MockedPipelineRequest
+### How to mock this module?
+
+It's easy to test functions that are using PipelineRequest. The only thing which is important is that tested function must return an original `Promise` when called. For example:
+```js
+// Action function.
+const fetchSomething = (...params) => () => {
+  // First, create a promise.
+  const promise = new PipelineRequest('pipelineName').dispatch();
+  // Then, define callbacks.
+  promise
+    .then(() => {
+      // Callback when pipeline request resolves.
+    })
+    .catch((e) => {
+      // Callback when pipeline request fails.
+    });
+ 
+  return promise;
+} 
+```
+
+When fetch action is implemented like in the example above, testing is pretty easy to implement. For example:
+```js
+// First import original fetch function 
+import fetchSomething from 'exampleFile';
+// Then, import mockPipelineRequestFactory
+import mockPipelineRequestFactory from '@shopgate/pwa-core/classes/PipelineRequest/mock';
+// Now, the PipelineRequest can be mocked.
+jest.mock(
+  '@shopgate/pwa-core/classes/PipelineRequest',
+  () => (
+    // Factory requires a callback function which will be called, always when PipelineRequest.dispatch is called.
+    mockedPipelineRequestFactory((mockedInstance, resolve, reject) => {
+      // Here can be placed the code which will decide if to resolve or reject a dispatch.
+      // For analysing purposes, mocked dispatch also passes the MockedPipelineRequest instance (this).
+      // This allows the test to analyse the internal data, like the effect of .setInput method.
+    })
+  )
+)
+```
+
+## Mocked methods and properties
+### Properties
+#### `static mockedDispatchResolver: function(this, resolve, reject)`
+Mock-specific property which contains a function which will be executed when .dispatch method is called.
+#### `name: string`
+Pipeline name which was passed when MockedPipelineRequest was created. Mock doesn't use it, but keeps it for assertion.
+#### `input: Object`
+Object which was passed by .setInput method. Defaults to an empty object.
```

Comment:
Grammer: **The o**bject which ... .setInput() ....

Reply:
done